### PR TITLE
Loris-MRI configurations moved to the front end from the $profile file: Redmine 8893

### PR DESCRIPTION
**Pull request by MounaSafiHarab**
Related to:
https://github.com/aces/Loris-MRI/pull/182
and
https://github.com/aces/Loris/pull/2649

These variables are now set in the Loris (front-end) Configuration module.

### CODE REVIEW

**Comment by MounaSafiHarab** [comment]
@driusan 
this one as well might need to disappear once repos are merged!

File: DB/DBI.pm [hunk 1]
```diff
@@ -54,4 +54,21 @@ sub connect_to_db
...
+sub getConfigSetting
+{
+    my ($dbh, $name) = @_;
+    my ($message,$query,$where) = '';
+    my $value = undef;
+
+    $where = " WHERE c.ConfigID=(Select cs.ID from ConfigSettings cs where cs.Name=?)";
+    $query = " SELECT c.Value FROM Config c";
+    $query = $query . $where;
+    my $sth = $$dbh->prepare($query);
```
Comment (gluneau) [inline DB/DBI.pm]: Not a SCALAR reference at ...DB/DBI.pm line 66.

Look for double dollar sign

File: DB/DBI.pm [hunk 2]
```diff
@@ -63,7 +63,7 @@ sub getConfigSetting
     $where = " WHERE c.ConfigID=(Select cs.ID from ConfigSettings cs where cs.Name=?)";
     $query = " SELECT c.Value FROM Config c";
     $query = $query . $where;
-    my $sth = $$dbh->prepare($query);
+    my $sth = $dbh->prepare($query);
```
Comment (cmadjar) [inline DB/DBI.pm]: should be 2 $ signs in front of $dbh :)

File: updateMRI_Upload.pl [hunk 3]
```diff
@@ -125,8 +125,10 @@
 ######generate an error#########################################
 ################################################################
 
-# fetch tarchiveLibraryDir from the config file
-my $tarchiveLibraryDir = $Settings::tarchiveLibraryDir;
+# fetch tarchiveLibraryDir from ConfigSettings in the database
+my $tarchiveLibraryDir = &DB::DBI::getConfigSetting(
+                            $dbh,'tarchiveLibraryDir'
+                            );
```
Comment (cmadjar) [inline updateMRI_Upload.pl]: update $dbh to be \$dbh

File: addScannerFields.pl [hunk 4]
```diff
@@ -60,8 +60,10 @@
 # establish database connection if database option is set
 my $dbh = &DB::DBI::connect_to_db(@Settings::db); print "Connecting to database.\n" if $verbose;
 
-# get $tarchiveLibraryDir from profile
-my $tarchiveLibraryDir = $Settings::tarchiveLibraryDir;
+# get $tarchiveLibraryDir from the ConfigSettings table
+my $tarchiveLibraryDir = &DB::DBI::getConfigSetting(
+                            $dbh,'tarchiveLibraryDir'
```
Comment (cmadjar) [inline addScannerFields.pl]: update $dbh to be \$dbh

File: MakeArchiveLocationRelative.pl [hunk 5]
```diff
@@ -41,16 +41,18 @@
...
+print "\n==> Successfully connected to database \n";
 
 ################################################################
-######### Establish database connection ########################
+#### This setting is in the ConfigSettings table   #############
 ################################################################
-my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
-print "\n==> Successfully connected to database \n";
+my $tarchiveLibraryDir = &DB::DBI::getConfigSetting(
+                            $dbh,'tarchiveLibraryDir'
```
Comment (cmadjar) [inline MakeArchiveLocationRelative.pl]: update $dbh to be \$dbh

File: LDZfix.pl [hunk 6]
```diff
@@ -50,13 +51,28 @@
...
+    exit;
+}
+################################################################
+######### Establish database connection ########################
+################################################################
+my $dbh = &DB::DBI::connect_to_db(@Settings::db);
+print "\n==> Successfully connected to database \n";
+
+my $get_dicom_info   = &DB::DBI::getConfigSetting(
+                        $dbh,'get_dicom_info'
```
Comment (cmadjar) [inline LDZfix.pl]: update $dbh to be \$dbh

File: DB/DBI.pm [hunk 7]
```diff
@@ -54,4 +54,21 @@ sub connect_to_db
...
+sub getConfigSetting
+{
+    my ($dbh, $name) = @_;
+    my ($message,$query,$where) = '';
+    my $value = undef;
+
+    $where = " WHERE c.ConfigID=(Select cs.ID from ConfigSettings cs where cs.Name=?)";
+    $query = " SELECT c.Value FROM Config c";
+    $query = $query . $where;
+    my $sth = $dbh->prepare($query);
```
Comment (cmadjar) [inline DB/DBI.pm]: Here should be $$dbh

File: DB/DBI.pm (hunk 2 above)
Comment (MounaSafiHarab) [inline DB/DBI.pm]: done! I changed it because it was not working for @gluneau 
(both ways work for me!)

File: DB/DBI.pm (hunk 7 above)
Comment (MounaSafiHarab) [inline DB/DBI.pm]: done

File: updateMRI_Upload.pl (hunk 3 above)
Comment (MounaSafiHarab) [inline updateMRI_Upload.pl]: done

File: addScannerFields.pl (hunk 4 above)
Comment (MounaSafiHarab) [inline addScannerFields.pl]: done

File: MakeArchiveLocationRelative.pl (hunk 5 above)
Comment (MounaSafiHarab) [inline MakeArchiveLocationRelative.pl]: done

File: LDZfix.pl (hunk 6 above)
Comment (MounaSafiHarab) [inline LDZfix.pl]: done

**Comment by MounaSafiHarab** [comment]
@gluneau 
could you please insert one more scan after fetching latest changes?
the `$` is back to its original `$$` as requested by @cmadjar, let me know if this breaks anything for you.

**Pull request merged**